Managers/RandomGenerator.gd

```diff
@@ -1,13 +1,9 @@
 extends Node
-
-
-var random: RandomNumberGenerator = null
+## Initializes randomness for the game.
 
 
 func _ready() -> void:
-	random = RandomNumberGenerator.new()
-	random.seed = randi()
-	
-	print("Random seed set to: " + str(random.seed))
+	var seed: int = randi()
+	seed(seed)
```

Comment:
does this set the seed globally for all random generator instances ? or just in a given scope ?

Reply:
This is for the entire game. So every random thing is going to work off of this seed. We probably won't want this in the long run, because I imagine we're going to want seeded randomness for stuff like map generation, but unseeded randomness for stuff like random card shuffling. This is helpful for now though because we can reproduce random stuff while debugging

Reply:
The seed is randomly set here, is there a way for us to decide the seed at launch ? Because we can't give a seed to the function so it sets this seed, it always uses `var seed: int = randi()`

Reply:
Yeah, we could just replace randi() with whatever number we want. I can add logic to the RandomGenerator that does this if we think it’s useful

Reply:
Well that would imply having to modify the code, it's probably better to have a structure like:
`func(given_seed: int = -1) -> void:
    var set_seed: int = given_seed
    if set_seed == -1:
        set_seed = randi()
    seed(set_seed)
    # print the set seed here
```
So `given_seed` defaults to -1 and we can give it a seed to put the one we want.

Reply:
Okay, I can add this